### PR TITLE
TST: split up pandas/tests/test_resample.py

### LINKED ISSUE #17806: TST/MAINT: split up test_resample.py to make it more understandable
As discussed in https://github.com/pandas-dev/pandas/pull/16153#issuecomment-315183678, test_resample.py grew sufficiently large (currently ~ 2.7 k sloc), so splitting it up should make it more understandable.

Proposal to split along the existing classes:

tests/resample (sub-dir)
|- common.py
|- test_resample_api.py
|- base.py
|- test_datetime.py
|- test_period.py
|- test_timedelta.py
|- test_resample_grouper.py
|- test_time_grouper.py

Moreover, readability of some test cases could be improved by using pytest features:
- Convert nested for loops and "quasi-parametrized" tests (multiple test methods calling the same test logic with different parameters) to make use of @pytest.mark.parametrize decorator
- Use pytest fixtures where appropriate

### PR DESCRIPTION
**Pull request by simonjayhawkins**
- [n/a ] precursor to #17806
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [n/a ] whatsnew entry

### CODE REVIEW

File: pandas/tests/resample/test_base.py [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+from datetime import datetime, timedelta
+
+import numpy as np
+import pytest
+
+from pandas.compat import range, zip
+from pandas.errors import AbstractMethodError
+
+import pandas as pd
+from pandas import DataFrame, Series
```
Comment (jreback) [inline pandas/tests/resample/test_base.py]: btw make sure you isort these new files and you may need to remove test_resample from setup.cfg

File: pandas/tests/resample/test_base.py [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+
+# The various methods we support
+downsample_methods = ['min', 'max', 'first', 'last', 'sum', 'mean', 'sem',
+                      'median', 'prod', 'var', 'ohlc', 'quantile']
+upsample_methods = ['count', 'size']
+series_methods = ['nunique']
+resample_methods = downsample_methods + upsample_methods + series_methods
+
+
+def _simple_ts(start, end, freq='D'):
```
Comment (jreback) [inline pandas/tests/resample/test_base.py]: let's rename these to something more readable and add a doc-string

File: pandas/tests/resample/test_base.py [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+                assert_index_equal(result.index, expected.index)
+                assert result.index.freq == expected.index.freq
+                assert_series_equal(result, expected, check_dtype=False)
+
+    def test_resample_empty_dataframe(self):
+        # GH13212
+        index = self.create_series().index[:0]
+        f = DataFrame(index=index)
+
+        for freq in ['M', 'D', 'H']:
```
Comment (jreback) [inline pandas/tests/resample/test_base.py]: in a future pass can parameterize things like this

File: pandas/tests/resample/test_timedelta_index.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
+# pylint: disable=E1101
+
+import numpy as np
```
Comment (jreback) [inline pandas/tests/resample/test_timedelta_index.py]: you can just name test_timedelta.py

File: pandas/tests/resample/test_period_index.py [hunk 5]
```diff
@@ -0,0 +1,739 @@
+# pylint: disable=E1101
+
+from datetime import datetime, timedelta
```
Comment (jreback) [inline pandas/tests/resample/test_period_index.py]: test_period is ok

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
i wouldn’t actually change the class structure / fixtures now
leave that for another pass

**Comment by simonjayhawkins** [comment]
@jreback 

> i wouldn’t actually change the class structure / fixtures now
leave that for another pass

OK, i'll just rename _simple_ts etc and leave it at that?

i've changed the checklist from closes to precursor to

File: pandas/tests/resample/test_base.py (hunk 1 above)
Comment (simonjayhawkins) [reply pandas/tests/resample/test_base.py]: done

File: pandas/tests/resample/test_base.py (hunk 3 above)
Comment (simonjayhawkins) [reply pandas/tests/resample/test_base.py]: OK

File: pandas/tests/resample/test_timedelta_index.py (hunk 4 above)
Comment (simonjayhawkins) [reply pandas/tests/resample/test_timedelta_index.py]: done

**Comment by jreback** [comment]
yeah a change like this is hard enough - first pass is just reorg minimally and rename

then can come back later and clean

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @simonjayhawkins nice split! sure happy to have a followup to address questions / simplify / use fixtures here

**Pull request merged**